servicemanager/smdownload.py

```diff
@@ -0,0 +1,45 @@
+import requests
+import time
+import sys
+
+
+def _report_hook(count, block_size, total_size):
+    global start_time
+    global last_update
```

Comment:
Why globals?
I wonder, as part of me would like to try parallelizing SM downloads at some point in the future.

Reply:
this was a cut/paste of the original progress meter. should be easy enough to refactor the globals out of it.
parallelizing the downloads would be cool, though rendering the progress might get a bit more complex